File: Python3/프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import deque, Counter, defaultdict
import heapq

def solution(participant, completion):
    answer = ''
    
    counter = Counter(participant)
    
    for c in completion:
        counter[c] -= 1
    
    for k, v in counter.items():
        if v > 0:
            answer = k
            break
    
    return answer

File: Java/프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        Map<String, Integer> map = new HashMap<>();
        
        for (String name : completion) {
            map.put(name, map.getOrDefault(name, 0) + 1);
        }
        
        for (String name : participant) {
            if (map.containsKey(name) && map.get(name) > 0) {
                map.put(name, map.get(name) - 1);
            } else {
                answer = name;
                break;
            }
        }
        
        return answer;
    }
}